File: 백준/Gold/11404. 플로이드/플로이드.js
const fs = require("fs");

const reader = {
  data: fs
    .readFileSync(process.platform === "linux" ? 0 : "input.txt", "utf-8")
    .toString()
    .trim()
    .split("\n"),
  
  cursor: 0,

  read() {
    return this.data[this.cursor++];
  },

  readInt() {
    return parseInt(this.read());
  },

  readIntArray() {
    return this.read().split(" ").map(s => parseInt(s));
  }
};

function solve() {
  const N = reader.readInt();
  const M = reader.readInt();

  const distance = Array.from({ length: N + 1 }, () => new Array(N + 1).fill(Infinity));

  for (let i = 1; i <= N; i++) {
    distance[i][i] = 0;
  }

  for (let i = 0; i < M; i++) {
    const [A, B, C] = reader.readIntArray();
    distance[A][B] = Math.min(distance[A][B], C);
  }

  for (let k = 1; k <= N; k++) {
    for (let i = 1; i <= N; i++) {
      for (let j = 1; j <= N; j++) {
        distance[i][j] = Math.min(distance[i][j], distance[i][k] + distance[k][j]);
      }
    }
  }

  let answer = "";

  for (let i = 1; i <= N; i++) {
    for (let j = 1; j <= N; j++) {
      const dist = distance[i][j] === Infinity ? 0 : distance[i][j];
      answer += `${dist} `;
    }
    answer += "\n";
  }

  return answer;
}

console.log(solve());

File: 백준/Gold/11404. 플로이드/플로이드.py
import sys

input = sys.stdin.readline
print = sys.stdout.write

nr_cities = int(input())
nr_routes = int(input())

distance = [[sys.maxsize for _ in range(nr_cities + 1)] for _ in range(nr_cities + 1)]

for i in range(1, nr_cities + 1):
    distance[i][i] = 0

for _ in range(nr_routes):
    s, e, w = map(int, input().split())
    if distance[s][e] > w:
        distance[s][e] = w

for k in range(1, nr_cities + 1):
    for s in range(1, nr_cities + 1):
        for e in range(1, nr_cities + 1):
            distance[s][e] = min(distance[s][e], distance[s][k] + distance[k][e])

for i in range(1, nr_cities + 1):
    for j in range(1, nr_cities + 1):
        if distance[i][j] == sys.maxsize:
            print("0 ")
        else:
            print(f"{distance[i][j]} ")
    print("\n")
